frontend/src/Users/Users.tsx

```diff
@@ -89,60 +118,206 @@ function EditUserDialog({ open, name, closeDialog }: EditUserProps): ReactElemen
     </TableRow>
   );
 }
 
 
 function Users() {
-  const { t } = useTranslation();
-  const [name, setName] = React.useState('');
+  const { t } = useTranslation('translation', { keyPrefix: "users" });
+  const [dialogData, setDialogData] = React.useState<CoreUserDialogData | null>(null);
+  const [allUsers, setAllUsers] = React.useState<SingleUserData[]>([]);
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+  const [informationalMessage, setInformationalMessage] = React.useState<string | null>(null);
+  const [fetchUsers, setFetchUsers] = React.useState<boolean>(true);
 
-  const openDialog = (fullName: string) => {
-    console.log('openDialog', fullName);
-    setName(fullName);
+  const reloadUsers = () => {
+    if (!fetchUsers) {
+      return;
+    }
+    axios.get(SERVER_ADDRESS + '/users', { withCredentials: true }).then((response) => {
+      if (response.status != 200) {
+        console.log('error', response);
+        return;
+      }
+      const allUsersApiData: object[] = response.data;
+      console.log('users data', allUsersApiData);
+      const apiAllUsers = allUsersApiData.map((apiUserData: any) => {
+        const user: SingleUserData = {
+          role: deserializeApiRole(apiUserData.role),
+          comment: apiUserData.comment,
+          id: apiUserData.id,
+          login: apiUserData.login,
+          name: apiUserData.name,
+          password: null,
+        }
+        return user;
+      });
+      apiAllUsers.sort((a, b) => b.id - a.id);  // Reverse to show new users on top
+      setAllUsers(apiAllUsers);
+      setFetchUsers(false);
+      setIsLoading(false);
+    })
+  }
+  React.useEffect(() => { reloadUsers() });
+
+  const openEditDialog = (user: SingleUserData) => {
+    setDialogData({ userData: user, message: null });
   };
 
-  const closeDialog = () => {
-    setName('');
+  const cancelDialog = () => {
+    setDialogData(null);
+  };
+
+  const saveDialog = (user: SingleUserData) => {
+    if (user.id == -1) {  // Creating new user
+      if (user.password === null || user.password.length === 0) {
+        setDialogData({ userData: user, message: t('dialog.password-required') });
+      } else if (user.login === '') {
+        setDialogData({ userData: user, message: t('dialog.login-required') });
+      } else if (allUsers.map((user) => user.login).includes(user.login)) {
+        setDialogData({ userData: user, message: t('dialog.user-exists') });
+      } else {
+        setDialogData(null);
+        setIsLoading(true);
+        axios.post(
+          SERVER_ADDRESS + '/users',
+          {
+            login: user.login,
+            password: user.password,
+            name: user.name,
+            role: UserType[user.role].toUpperCase(),
+            comment: user.comment,
+          },
+          { withCredentials: true },
+        ).then((response) => {
+          if (response.status != 200) {
```

Comment:
If response has not 200 code, code in ```.then``` won't execute, you need ```.catch(error)``` for handling errors